tokenizer/udic.go

```diff
@@ -0,0 +1,127 @@
+// Copyright 2016 ikawaha
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// 	You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tokenizer
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"os"
+	"sort"
+	"strings"
+
+	"github.com/ikawaha/kagome.ipadic/internal/dic"
+)
+
+// UserDic represents a user dictionary.
+type UserDic struct {
+	dic *dic.UserDic
+}
+
+// NewUserDic build a user dictionary from a file.
+func NewUserDic(path string) (UserDic, error) {
+	f, err := os.Open(path)
+	if err != nil {
+		return UserDic{}, err
+	}
+	defer f.Close()
```

Comment:
__[gometalinter]__
* [errcheck] error return value not checked (defer f.Close())

<!-- Comment by [SideCI](https://sideci.com) -->